### PR TITLE
fix: Allow docker image not to have tags

### PR DESCRIPTION
**Pull request by JulesFaucherre**

### CODE REVIEW

**Review by rlegan — APPROVED** [review]

File: pkg/services/testdata/noErrors.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ orbs:
 executors:
   go:
     docker:
-      - image: cimg/go:latest
+      - image: cimg/go:1.20.3
```
Comment (rlegan) [inline pkg/services/testdata/noErrors.yml:11]: we should keep at least one test with `latest` imo

File: pkg/parser/validate/validate.go [hunk 2]
```diff
@@ -1,13 +1,19 @@
...
 
 	"go.lsp.dev/protocol"
 )
 
+type ValidateAPIs struct {
+	DockerHub dockerhub.DockerHubAPI
+}
+
 type Validate struct {
+	APIs        ValidateAPIs
```
Comment (rlegan) [inline pkg/parser/validate/validate.go:16]: why plural ? we can have multiple apis ?

File: pkg/dockerhub/exists.go [hunk 3]
```diff
@@ -17,20 +19,13 @@ func DoesImageExist(namespace, image, tag string) bool {
...
-		// in that case, the image is valid as long as a repo with the name exists.
-
-		url = url.JoinPath(
-			fmt.Sprintf("/tags/%s", tag),
-		)
-	}
-
 	req, err := http.NewRequest("GET", url.String(), nil)
+
+	fmt.Printf("url.String() = %+v\n", url.String())
```
Comment (rlegan) [inline pkg/dockerhub/exists.go:28]: to delete

File: pkg/parser/validate/validate.go (hunk 2 above)
Comment (JulesFaucherre) [reply pkg/parser/validate/validate.go:16]: Yes, atm there is only the DockerHub API we can imagine having the same logic for the CCI API

**Pull request merged**